src/main/java/org/cirdles/topsoil/dataset/DatasetResource.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.dataset;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.cirdles.topsoil.app.dataset.writer.TSVDatasetWriter;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class DatasetResource {
```

Comment:
You should really extract `DatasetResource` as an interface. That'll help clear up separation of concerns.